packages/fx-core/src/component/driver/oauth/utility/utility.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { SpecParser } from "@microsoft/m365-spec-parser";
+import { getAbsolutePath } from "../../../utils/common";
+import { DriverContext } from "../../interface/commonArgs";
+import { CreateOauthArgs } from "../interface/createOauthArgs";
+import { isApiKeyEnabled, isMultipleParametersEnabled } from "../../../../common/featureFlags";
+import { OpenAPIV3 } from "openapi-types";
+import { isEqual } from "lodash";
+import { maxDomainPerApiKey } from "./constants";
+import { OauthDomainInvalidError } from "../error/oauthDomainInvalid";
+import { OauthFailedToGetDomainError } from "../error/oauthFailedToGetDomain";
+import { OauthAuthInfoInvalid } from "../error/oauthAuthInfoInvalid";
+
+export interface OauthInfo {
+  domain: string[];
+  authorizationEndpoint: string;
+  tokenExchangeEndpoint: string;
+  tokenRefreshEndpoint?: string;
+  scopes: string[];
+}
+
+interface AuthInfo {
+  authorizationUrl: string;
+  tokenUrl: string;
+  refreshUrl?: string;
+  scopes: string[];
+}
+
+export async function getandValidateOauthInfoFromSpec(
+  args: CreateOauthArgs,
+  context: DriverContext,
+  actionName: string
+): Promise<OauthInfo> {
+  const absolutePath = getAbsolutePath(args.apiSpecPath, context.projectPath);
+  const parser = new SpecParser(absolutePath, {
+    allowBearerTokenAuth: isApiKeyEnabled(), // Currently, API key auth support is actually bearer token auth
+    allowMultipleParameters: isMultipleParametersEnabled(),
```

Comment:
isApiKeyEnabled() feature flag is used for SME and already been set to true, I think we should use another one for copilot.

Reply:
Updated, thanks!